created episode page for 35 (#97)
rustacean-station/rustacean-station.org: _episodes/035-daniel-stenberg.md

---
title: "Rust in cURL"

> date: 2021-09-03T23:00:00Z

Reviewer: This is a fairly late time on Friday (https://everytimezone.com/s/06ed398f). I would probably make this an earlier time, just because podcast clients sometimes get confused for episodes with times too far into the future.

file: https://audio.rustacean-station.org/file/rustacean-station/rustacean-station-e035-daniel-stenberg.mp3
duration: "55:25"
length: "53290745"
#reddit: (leave blank on initial publish, amend with link and uncomment this line after Reddit thread has been posted)
---
#  Rust in cURL  

First time guest host, Allen Wyma talks with Daniel, the original author of cURL, about using Rust in cURL.

> - cURL is a command line tool and library for transferring data with URLs.

Reviewer: I would link to https://curl.se/ directly here, not just in the resources list at the bottom.

> - cURL and its data transfer core, libcurl is all written in C, which is known to be not memory safe. 

Reviewer: ```suggestion
- cURL, and its data transfer core libcurl, are both written in C, which is known to be not memory safe. 
```

- While it is almost impossibe to rewrite it into another language, offering a third-party library written in Rust could take a further step forward.

## Contributing to Rustacean Station

Rustacean Station is a community project; get in touch with us if you'd like to suggest an idea for an episode or offer your services as a host or audio editor!

- Twitter: [@rustaceanfm](https://twitter.com/rustaceanfm)
- Discord: [Rustacean Station](https://discord.gg/cHc3Gyc)
- Github: [@rustacean-station](https://github.com/rustacean-station/)
- Email: [hello@rustacean-station.org](mailto:hello@rustacean-station.org)

> ## Highlights

Reviewer: We tend to instead have a "Timestamps & referenced resources" section in which we list timestamps for particular topics, and also links to relevant resources _for_ that topic if any. You can see an example [here](https://raw.githubusercontent.com/rustacean-station/rustacean-station.org/master/_episodes/034-rust-1.52-1.53.md). This gives listeners an easy way to skip to the content they care most about.

- Architecture of cURL & libcurl 
- What brought Daniel to Rust? How long has he been using Rust?
- What language was Daniel working in before that? 
- Third party libraries written in Rust?
- What benefits does Rust bring to cURL?

## Resources 
 - [Curl](https://curl.se/)
 - [Daniel's Blog](https://daniel.haxx.se/blog/2021/08/09/nocais-apology/)

>  - [Project Gemini](https://gemini.circumlunar.space/)

Reviewer: We generally include a Credits section at the bottom of the notes. See [here](https://raw.githubusercontent.com/rustacean-station/rustacean-station.org/master/_episodes/034-rust-1.52-1.53.md) for an example.

---
title: "Rust in cURL"

> date: 2021-09-03T10:00:00Z

Reviewer: ```suggestion
date: 2021-09-03T15:30:00Z
```

## Contributing to Rustacean Station

Rustacean Station is a community project; get in touch with us if you'd like to suggest an idea for an episode or offer your services as a host or audio editor!

- Twitter: [@rustaceanfm](https://twitter.com/rustaceanfm)
- Discord: [Rustacean Station](https://discord.gg/cHc3Gyc)
- Github: [@rustacean-station](https://github.com/rustacean-station/)
- Email: [hello@rustacean-station.org](mailto:hello@rustacean-station.org)

## Highlights

- Architecture of cURL & libcurl 
- What brought Daniel to Rust? How long has he been using Rust?
- What language was Daniel working in before that? 
- Third party libraries written in Rust?
- What benefits does Rust bring to cURL?

## Resources 
 - [Curl](https://curl.se/)
 - [Daniel's Blog](https://daniel.haxx.se/blog/2021/08/09/nocais-apology/)
 - [Project Gemini](https://gemini.circumlunar.space/)

## Timestamps 
- [@05:10] - [cURL 7.78.0](https://curl.se/download.html)
- [@07:44] - Implementing Protocol
- [@09:25] - HTTP/3
- [@13:30] - Architecture of cURL & libcurl

> - [@17:40] - Hybird library 

Reviewer: ```suggestion
- [@17:40] - cURL as a hybrid library 
```
